Assets/Scripts/Skills/SkillSystem.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Timers;
+using UnityEngine;
+
+public class SkillSystem
+{
+    public event Action<float> IsHeal;
+    public event Action<float> IsBoostSpeed;
+    public event Action<float> IsIncreaseDamage;
+
+    
+    private HealthSystem _healthSystem;
+    private PlayerCharacter _player;
+    private Func<Weapon> _getWeapon;
+    private Weapon _currentWeapon;
+
+    //Characteristics for skill object
+    private const float _mSecForHealSkill = 1000;
+    private const float _mSecForBoostSkill = 2000;
+    private const float _mSecForDamageSkill = 5000;
+
+    private bool _continuesHealSkill;
+    private bool _continuesBoostSkill;
+    private bool _continuesDamageSkill;
+
+    public SkillSystem(HealthSystem healthSystem, PlayerCharacter player, Func<Weapon>  getWeapon)
+    {
+        _healthSystem = healthSystem;
+        _player = player;
+        _getWeapon = getWeapon;
+        _currentWeapon = getWeapon.Invoke();
+    }
```

Comment:
Fix errors in other places to match this new constructor
```suggestion
    public SkillSystem(HealthSystem healthSystem, PlayerCharacter player, WeaponSystem weaponSystem)
    {
        _healthSystem = healthSystem;
        _player = player;
        _weaponSystem = weaponSystem;
        _currentWeapon = _weaponSystem.GetCurrentWeapon();
    }
```